File: Biweekly Contest 123/Q4.java
https://leetcode.com/contest/biweekly-contest-123/problems/find-the-number-of-ways-to-place-people-ii/
3027. Find the Number of Ways to Place People II

class Solution {
    public int numberOfPairs(int[][] points) {
        int n=points.length;
        //sorting the array based on 1st index and if same ,then sort the 2nd index which is larger
        Arrays.sort(points,new Comparator<int[]>(){
            public int compare(int[] a,int[] b){
                if(a[0]==b[0])return Integer.compare(b[1],a[1]);
                return Integer.compare(a[0],b[0]);
            }
        });
        int ans=0;
        
        for(int i=0;i<n;i++){//choosing the top element
            int max_so_far=Integer.MIN_VALUE;//try to choose the point which has max y than previous accepted
            for(int j=i+1;j<n;j++){//selecting whether the current element can be paired
                if(points[i][1]>=points[j][1]){
                    if(points[j][1]>max_so_far){
                        ans++;
                        max_so_far=points[j][1];//making the current y as max
                    }
                }
            }
        }
        
        return ans;
            
    }
}

File: Biweekly Contest 123/Q4.cpp
class Solution {
public:
    static bool cmp(vector<int> &a,vector<int> &b){
        if(a[0]==b[0])return a[1]>b[1];
        return a[0]<b[0];
    }
    int numberOfPairs(vector<vector<int>>& points) {
        int ans=0;
        sort(points.begin(),points.end(),cmp);
        int n=points.size();
        for(int i=0;i<n;i++){
            int min_so_far=INT_MAX;
            for(int j=i-1;j>=0;j--){
                if(points[i][1]<=points[j][1]){
                    if(points[j][1]<min_so_far){
                        ans++;
                        min_so_far=points[j][1];
                    }
                }
            }
        }
        return ans;
    }
};